test/functional/change_stream_tests.js

```diff
@@ -152,11 +154,13 @@ describe('Change Streams', function() {
 
         var thisChangeStream1, thisChangeStream2, thisChangeStream3;
 
-        theCollection1
-          .insert({ a: 1 })
-          .then(function() {
+        setTimeout(() => {
```

Comment:
`setTimeout` with no actual timeout? Do you mean `setImmediate` or `process.nextTick`?  It looks like this happens a few more times, I'll just leave this note here with an understanding that we should consider it for the full file.

Reply:
I think any one of the three will work. `setTimeout` is lower priority than `setImmediate` and `process.nextTick`, so I'd prefer it to be as delayed as possible.